src/missing-types.d.ts

```diff
@@ -91,40 +98,233 @@ declare module '@higlass/services' {
     };
 }
 
-// TODO(06-21-22)
 declare module '@higlass/tracks' {
+    // TODO(2022-06-28): type out `BarTrack`
+    type Track = any;
+    export const BarTrack: Track;
+
+    import type { ScaleContinuousNumeric } from 'd3-scale';
+    import type * as PIXI from 'pixi.js';
+
+    type Scale = ScaleContinuousNumeric<number, number>;
+
+    type Handler = (data: any) => void;
+
+    type Subscription = { event: string; handler: Handler };
+
+    type PubSub = {
+        publish(msg: string, data: any): void;
+        subscribe(msg: string, handler: Handler): Subscription;
+        unsubscribe(msg: string): void;
+    };
+
+    type TrackOptions = Record<string, unknown>;
+
     type Track = any;
-    export declare const BarTrack: Track;
-    export declare const PixiTrack: Track;
+    export const BarTrack: Track;
+
+    export type Context<Options> = {
+        id: string;
+        viewUid: string;
+        pubSub: PubSub;
+        scene: PIXI.Graphics;
+        dataConfig: DataConfig;
+        dataFetcher: DataFetcher;
+        getLockGroupExtrema(): [min: number, max: number] | null;
+        handleTilesetInfoReceived(tilesetInfo: TilesetInfo): void;
+        animate(): void;
+        svgElement: HTMLElement;
+        isValueScaleLocked(): boolean;
+        onValueScaleChanged(): void;
+        onTrackOptionsChanged(newOptions: Options): void;
+        // TODO: not sure what the props are here
+        onMouseMoveZoom(props: unknown): void;
```

Comment:
This function seems to be used in both [HorizontalTiled1DPixiTrack.js](https://github.com/higlass/higlass/blob/54f5aae61d3474f9e868621228270f0c90ef9343/app/scripts/HorizontalTiled1DPixiTrack.js#L201) and [HeatmapTiledPixiTrack.js](https://github.com/higlass/higlass/blob/54f5aae61d3474f9e868621228270f0c90ef9343/app/scripts/HeatmapTiledPixiTrack.js#L224), and the type seem to be the following (ignore the names):

```ts
interface commonProps {
  trackId: string,
  data: number,
  absX: number,
  absY: number,
  relX: number,
  relY: number,
  dataX: number,
  dataY: number,
}
interface commonProps1D extends commonProp {
  orientation: '1d-horizontal' | '1d-vertical';
}
interface commonProps2D extends commonProp {
  orientation: '2d',
  dataLens: Float32Array,
  dim: number,
  toRgb: [number, number, number, number],
  center: [number, number],
  xRange: [number, number],
  yRange: [number, number],
  isGenomicCoords: boolean,
}
type props = commonProps1D | commonProps2D;
```

Reply:
thanks so much for looking into this! The only type I changed was `dataLens` to `ArrayLike<number>` since it is assigned differently in the component.